pkg/apiclient/application/forwarder_overwrite.go

```diff
@@ -13,7 +18,32 @@ import (
 
 func init() {
 	forward_ApplicationService_PodLogs_0 = http.StreamForwarder
-	forward_ApplicationService_PodLogs_1 = http.StreamForwarder
+	forward_ApplicationService_PodLogs_1 = func(ctx context.Context, mux *runtime.ServeMux, marshaler runtime.Marshaler, w gohttp.ResponseWriter, req *gohttp.Request, recv func() (proto.Message, error), opts ...func(context.Context, gohttp.ResponseWriter, proto.Message) error) {
+		if req.URL.Query().Get("download") == "true" {
+			w.Header().Set("Content-Type", "application/octet-stream")
+			fileName := "log"
+			if container := req.URL.Query().Get("container"); len(container) > 0 {
+				fileName = container
+			}
```

Comment:
Can we do some sanity check on the `container` parameter that was requested here? Like, that it does not contain any special characters here? We take that value and send it back to the client in a header, so we really should do some checking on its value :)

Container name is validated by Kubernetes against https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names, so we could use following method (RE is same that K8s uses):

```go
func isValidResourceName(name string) bool {
	re := regexp.MustCompile("^[a-z0-9]([-a-z0-9]*[a-z0-9])?$")
	return len(name) < 64 && re.MatchString(name)
}
```
I haven't looked yet if there is a similar method in our code base already, but it might make sense to put this method somewhere shared, too :) Or check if K8s APIs provide such a method already.

Reply:
Good point. I don't want one more security advisory :) fixed.

Reply:
Thanks!